advisor/src/main/kotlin/advisors/NexusIq.kt

```diff
@@ -88,11 +93,18 @@ class NexusIq(
             }
 
             try {
-                val componentDetails =
-                    execute(withContext(advisorDispatcher) {
-                        service.getComponentDetails(NexusIqService.ComponentsWrapper(components))
-                    }).componentDetails
-                        .associateBy { it.component.packageUrl.substringBefore("?") }
+                val chunks = components.chunked(REQUEST_CHUNK_SIZE)
+
+                val componentDetails = mutableMapOf<String, NexusIqService.ComponentDetails>()
+                chunks.forEach { component ->
+                    val requestResults =
+                        execute(withContext(advisorDispatcher) {
+                            service.getComponentDetails(NexusIqService.ComponentsWrapper(component))
+                        }).componentDetails
+                            .associateBy { it.component.packageUrl.substringBefore("?") }
+
+                    componentDetails += requestResults
+                }
```

Comment:
How about using the chance to slightly rewrite this for more readabilty (IMO):

                val componentDetails = mutableMapOf<String, NexusIqService.ComponentDetails>()

                components.chunked(REQUEST_CHUNK_SIZE).forEach { component ->
                    val serviceCall = withContext(advisorDispatcher) {
                        service.getComponentDetails(NexusIqService.ComponentsWrapper(component))
                    }

                    val requestResults = execute(serviceCall).componentDetails.associateBy {
                        it.component.packageUrl.substringBefore("?")
                    }

                    componentDetails += requestResults
                }